### ISSUE TITLE
False positive `UnusedImport` on `inline val`

### ISSUE THREAD

**Issue by mgroth0** [detekt/detekt#7815]
* Version of detekt used: commit da8b164c5da3a2c8017452355c1754df7344c8a7

```kotlin
package my.package

inline val myVal get() = 1
```

```kotlin
import my.package.myVal // reported as `UnusedImport`

fun main() {
    println(myVal)
}
```

**Comment by cortinico** [comment]
This seems like an edge case the rule doesn't cover + a potentially good/simple contribution for someone that wants to pick this up

**Comment by maxweilun1989** [comment]
@cortinico I’d like to give it a try. Could you assign this to me?

**Comment by maxweilun1989** [comment]
@mgroth0  Sorry, I couldn't reproduce the issue. I switch to commit da8b164c5d, and wrote the test below, but no `UnusedImport` was reported. 
```kotlin
    @Test
    fun `should not reports when import an inline val`(){
        val mainFile = """
            import x.y.myVal 

            fun test() {
                println(myVal)
            }
        """.trimIndent()

        val additionalFile = """
            package x.y

            inline val myVal get() = 1
        """.trimIndent()

        assertThat(subject.lintWithContext(env, mainFile, additionalFile)).isEmpty()
    }
```

Did I get you wrong?

**Comment by AncutaIoan** [comment]
Anyone still working onto this one?

**Comment by BraisGabin** [comment]
I don't think so. If you want to take it, you can.

**Comment by AncutaIoan** [comment]
So just as an update, added a sanity check test for this and for the provide code scenario is working as expected, will try now to look into the generated docs behavior

**Comment by AncutaIoan** [comment]
PR : https://github.com/detekt/detekt/pull/8287

**Comment by BraisGabin** [comment]
Closing because that pr proves that it works. Probably there is something else that make this happen. If you can provide a reproducer we can reopen it.
